rsconf/component/rsiviz.py

```diff
@@ -19,27 +19,33 @@ def internal_build_compile(self):
         jc, z = self.j2_ctx_init()
         z._run_u = jc.rsconf_db.run_u
         self.j2_ctx_pksetdefault(
-            PKDict(pykern=PKDict(pkconfig=PKDict(channel=jc.rsconf_db.channel)))
+            PKDict(
+                pykern=PKDict(
```

Comment:
I think this is problematic. Rather, it should be cascaded explicitly, e.g. rsiviz.server_port gets assigned to .pykern.pkasyncio.server_port. Otherwise, you can only have one pkasyncio service per host. I would also probably define rsiviz.index_iframe_port and cascade that to rsiviz.pkcli.service....

Reply:
> Otherwise, you can only have one pkasyncio service per host

The value is set in /srv/rsiviz/env so another service could set a different port in its /srv/foo/env. We could only have one rsiviz service per host (because their ports would conflict) but that doesn't seem very problematic because, to my knowledge, we don't have a way of running multiple rsiviz services on a host.

But, cascade makes sense ff62898

Reply:
It shouldn't be j2_ctx_pksetdefault. This should be a pkudpate. The problem is that if someone already has set any of these pykern values in the yaml, it will not override with the values being set here.

Reply:
3cba232